backend/api/silenced.go

```diff
@@ -0,0 +1,149 @@
+package api
+
+import (
+	"context"
+	"fmt"
+
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+	"github.com/sensu/sensu-go/backend/authorization"
+	"github.com/sensu/sensu-go/backend/store"
+)
+
+// SilencedClient is an API client for silencing checks.
+type SilencedClient struct {
+	store store.SilencedStore
+	auth  authorization.Authorizer
+}
+
+// NewSilencedClient creates a new SilencedClient, given a store and authorizer.
+func NewSilencedClient(store store.SilencedStore, auth authorization.Authorizer) *SilencedClient {
+	return &SilencedClient{
+		store: store,
+		auth:  auth,
+	}
+}
+
+// UpdateSilenced updates a silenced entry, if authorized.
+func (s *SilencedClient) UpdateSilenced(ctx context.Context, silenced *corev2.Silenced) error {
+	if err := silenced.Validate(); err != nil {
```

Comment:
Either in the mutation or the client we'll want to prepare the entry.

```suggestion
	silenced.Prepare(ctx)
```

Reply:
Ah, thanks!